src/stream/src/executor/source/source_executor.rs

```diff
@@ -610,6 +653,58 @@ impl<S: StateStore> Debug for SourceExecutor<S> {
     }
 }
 
+struct WaitEpochWoker<S: StateStore> {
+    wait_epoch_rx: UnboundedReceiver<(Epoch, HashMap<SplitId, String>)>,
+    state_table_handler: Arc<Mutex<SourceStateTableHandler<S>>>,
+}
+
+impl<S: StateStore> WaitEpochWoker<S> {
+    pub async fn run(mut self) {
+        tracing::debug!("wait epoch worker start success");
+        loop {
+            // poll the rx and wait for the epoch commit
+            match self.wait_epoch_rx.recv().await {
+                Some((epoch, updated_offsets)) => {
+                    let state_store_guard = self.state_table_handler.lock().await;
+                    tracing::debug!("start to wait epoch {}", epoch.0);
+                    let ret = state_store_guard.state_store.wait_epoch(epoch.0).await;
+                    drop(state_store_guard);
+
+                    match ret {
+                        Ok(()) => {
+                            tracing::debug!(epoch = epoch.0, "wait epoch success");
+                            // cdc source only has one split
+                            assert_eq!(1, updated_offsets.len());
+                            let (split_id, offset) = updated_offsets.into_iter().next().unwrap();
+                            let source_id: u64 = u64::from_str(split_id.as_ref()).unwrap();
+                            // notify cdc connector to commit offset
+                            match jni_source::commit_cdc_offset(source_id, offset.clone()) {
+                                Ok(_) => {}
+                                Err(e) => {
+                                    tracing::error!(
+                                        error = ?e.as_report(),
```

Comment:
```suggestion
                                        error = %e.as_report(),
```

`?` will include the backtrace. I'm afraid this is not expected here. Same for other occurrences.

Reply:
fixed